CharlieBackend.Data/Repositories/Impl/StudentGroupRepository.cs

```diff
@@ -114,5 +116,16 @@ public async Task<List<MentorStudyGroupsDto>> GetMentorStudyGroups(long id)
             return await _applicationContext.StudentsOfStudentGroups.Where(s => s.StudentGroupId == id)
                 .Select(s => s.StudentId).ToListAsync();
         }
+
+        public async Task<IList<StudentGroup>> GetStudentGroupsByDateAsync(DateTime startDate,DateTime finishDate)
+        {
+            return await _applicationContext.StudentGroups
+                .AsNoTracking()
+                .Where(x => x.StartDate < finishDate && x.StartDate >= startDate || 
+                       x.FinishDate > startDate && x.FinishDate <= finishDate)
+                .Where(x => x.Course.IsActive)
+                .OrderBy(x => x.StartDate)
+                .ToListAsync();
+        }
```

Comment:
please provide examples that describe the logic of the filter, otherwise there is no way to determine the correctness of the condition in the given code.